implementations/rust/ockam/ockam_api/src/nodes/service/in_memory_node.rs

```diff
@@ -58,13 +58,14 @@ impl Drop for InMemoryNode {
         // because in that case they can be restarted
         if !self.persistent {
             executor::block_on(async {
-                // We need to recreate the CliState here to make sure that
-                // we get a fresh connection to the database (otherwise this code blocks)
-                let cli_state = CliState::create(self.cli_state.dir()).await.unwrap();
-                cli_state
-                    .remove_node(&self.node_name)
-                    .await
-                    .unwrap_or_else(|e| panic!("cannot delete the node {}: {e:?}", self.node_name));
+                let result = self.cli_state.remove_node(&self.node_name).await;
+                if let Err(err) = result {
+                    // code: 1032 maps to SQLITE_READONLY_DBMOVED - meaning the database has been
+                    // moved to another directory, most likely already deleted
+                    if !err.to_string().contains("code: 1032") {
+                        panic!("cannot delete the node {}: {err:?}", self.node_name);
```

Comment:
Should we really panic here?

Reply:
I've changed the panic into an error, it shouldn't happen, but even if it does, it wouldn't result in a catastrophic effect